stat/combin/combin.go

```diff
@@ -185,6 +185,98 @@ func nextCombination(s []int, n, k int) {
 	}
 }
 
+// PermutationIndex returns the index of the given permutation.
+//
+// The functions PermutationIndex and IndexToPermutation define a bijection
+// between the integers and the permutations of the integers [0, len(perm)-1].
+// PermutationIndex returns the inverse of IndexToPermutation.
+//
+// PermutationIndex panics if perm is not a permutation of the first len(perm)
+// non-negative integers.
+func PermutationIndex(perm []int) int {
+	// Note(btracey): This is an n^2 algorithm, but factorial increases
+	// very quickly (25! overflows int64) so this is not a problem in
+	// practice.
+
+	check := map[int]struct{}{}
+	for _, v := range perm {
+		if 0 <= v || v >= len(perm) {
```

Comment:
`0 <= v` should be `v < 0`?

Reply:
Also, `len(perm) <= v`.

Reply:
I guess an easier way to spell it might be:
```go
if !(0 <= v && v < len(perm)) {
```

Reply:
We normally write value out of range as `v < 0 || len(perm) <= v`.